### PR TITLE
Binary to Decimal Number in C

### PR DESCRIPTION
**Pull request by imsushant12**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

YES

### Description

This program will take a binary number as input and convert it into decimal number.

### Checklist

- [ YES ] I've read the contribution guidelines.
- [ YES ] I've checked the issue list before deciding what to submit.
- [ YES ] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

https://github.com/TesseractCoding/NeoAlgo/issues/6774

### CODE REVIEW

File: C/README.md [hunk 1]
```diff
@@ -114,6 +114,7 @@ _add list here_
 ## Mathematics
 
 - [Binary Exponentiation](math/Binary_Exponentiation.c)
+- [Binary to Decimal](math/binary_to_decimal.c)
```
Comment (ankitaggarwal23) [inline C/README.md]: Rewrite it as `- [Binary to Decimal Conversion] `

Shift the problem to other file location as this is not a math problem.

File: C/math/binary_to_decimal.c [hunk 2]
```diff
@@ -0,0 +1,44 @@
+/*
+AIM - To take binary number from the user 
+      and convert it into binary number.
+*/
+#include <stdio.h>
+#include <stdlib.h>
+
+/*Function decleration*/
+int BtoD (char s[]);
```
Comment (ankitaggarwal23) [inline C/math/binary_to_decimal.c]: Give meaningful function name

File: C/math/binary_to_decimal.c [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    int i,d=0;
+    for(i=0;s[i];i++)
+    {
+        if(s[i]=='0')
+            d=d*2+0;
+        else
+            d=d*2+1;
+    }
+    return(d);
+}
```
Comment (ankitaggarwal23) [inline C/math/binary_to_decimal.c]: Don't use single letter for variable names.

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

File: C/math/binary_to_decimal.c (hunk 3 above)
Comment (imsushant12) [reply C/math/binary_to_decimal.c]: done

File: C/math/binary_to_decimal.c (hunk 2 above)
Comment (imsushant12) [reply C/math/binary_to_decimal.c]: done

File: C/README.md (hunk 1 above)
Comment (imsushant12) [reply C/README.md]: done sir, please review.

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**